### PR TITLE
Driver view pending requests

### PR DESCRIPTION
**Pull request by MichaelParadis**
Finished US 05.02.01: As a driver, I want to view a list of things I have accepted that are pending, each request with its description, and locations. Addresses issue #84

### CODE REVIEW

File: app/src/test/java/comcmput301f16t01/github/carrier/RequestControllerTest.java [hunk 1]
```diff
@@ -1,10 +1,63 @@
...
+import static org.junit.Assert.*;
 
 public class RequestControllerTest {
     @Test
     public void addRequestTest() {
         // TODO write tests to test adding a request via elastic search
     }
+
+    @Test
+    public void addOfferingDriverTest() {
```
Comment (kieter) [inline app/src/test/java/comcmput301f16t01/github/carrier/RequestControllerTest.java:13]: I like tests.

**Comment by Hreherch** [comment]
I liked the general vibe of this one :)
I've noticed with this one that we are starting to have code duplication, so I'm wondering how we could avoid it. But that's more of an enhancement issue.

**Review by kieter — COMMENTED** [review]
This looks good to me. Very pretty. Only thing is the app crashes when you tap the requests in the rider tab which has nothing to do with your code Mike 😛.

File: app/src/test/java/comcmput301f16t01/github/carrier/RequestControllerTest.java [hunk 2]
```diff
@@ -1,10 +1,63 @@
...
+        RequestList rl = RequestController.getInstance();
+        Request request = new Request(rider, new Location(), new Location());
+        rl.add(request);
+        // Test that the empty requests work
+        assertTrue("Driver has offered request when they shouldn't", rc.getOfferedRequests(offeringDriver).size() == 0);
+        rc.addDriver(request, offeringDriver);
+        // Make sure the driver is added the request
+        assertTrue("Get offered requests is not returning the correct request", rc.getOfferedRequests(offeringDriver).contains(request));
+        // Make sure that the driver is only added to one request.
+        assertTrue("The driver is added to more than one reqeust", rc.getOfferedRequests(offeringDriver).size() == 1);
```
Comment (kieter) [inline app/src/test/java/comcmput301f16t01/github/carrier/RequestControllerTest.java:37]: 👍

File: app/src/test/java/comcmput301f16t01/github/carrier/RequestControllerTest.java [hunk 3]
```diff
@@ -1,10 +1,63 @@
...
+        // Make sure requests aren't added for the wrong rider.
+        assertTrue("rider1 has an open request when they shouldn't.", rc.getRequests(rider1).size() == 0);
+        Request request1 = new Request(rider1, new Location(), new Location());
+        rl.add(request1);
+        // Make sure the request is added to rider1's requests.
+        assertTrue(rl == RequestController.getInstance());
+        assertTrue(rl.contains(request1));
+        assertTrue(request1.getRider() == rider1);
+        assertTrue("Request not added for rider1", rc.getRequests(rider1).contains(request1));
+        // Make sure only that request is added.
```
Comment (kieter) [inline app/src/test/java/comcmput301f16t01/github/carrier/RequestControllerTest.java:58]: 💯

**Review by kieter — APPROVED** [review]

**Pull request merged**